packages/zksend/src/experimental/wallet/index.ts

```diff
@@ -116,16 +121,41 @@ export class StashedWallet implements Wallet {
 
 	constructor({
 		name,
-		origin = DEFAULT_STASHED_ORIGIN,
+		origin,
+		useDynamicMetadata = true,
 	}: {
 		name: string;
 		origin?: string;
+		useDynamicMetadata?: boolean;
 		chain?: SuiChain;
 	}) {
 		this.#accounts = [];
 		this.#events = mitt();
-		this.#origin = origin;
+		this.#origin = origin || DEFAULT_STASHED_ORIGIN;
 		this.#name = name;
+		this.#useDynamicMetadata = useDynamicMetadata;
+		if (this.#useDynamicMetadata) {
```

Comment:
IMO these should be fetched as part of the `register` call so that we can disable the actual registration itself. We probably want to disable this by default, and only enable it when the HTTP call comes back (for the initial release of the wallet at least)